trestlebot/entrypoints/create_ssp.py

```diff
@@ -0,0 +1,143 @@
+            help="Comma-separated list of component definitions to include in the SSP",
+        )
+        self.parser.add_argument(
+            "--leveraged-ssp",
+            required=False,
+            type=str,
+            help="Provider SSP to leverage for the new SSP.",
+        )
+        self.parser.add_argument(
+            "--markdown-path",
+            required=True,
+            type=str,
+            help="Path to create markdown files in.",
+        )
+        self.parser.add_argument(
+            "--yaml-header-path",
+            required=False,
+            type=str,
+            help="Optionally set a path to a YAML file for custom SSP Markdown YAML headers.",
+        )
+        self.parser.add_argument(
+            "--version",
+            required=False,
+            type=str,
+            help="Optionally set the SSP version.",
+        )
+        self.parser.add_argument(
+            "--ssp-index-path",
+            required=False,
+            type=str,
+            default="ssp-index.json",
+            help="Optionally set the path to the SSP index file.",
+        )
+
+    def run(self, args: argparse.Namespace) -> None:
+        """Run the entrypoint."""
+        exit_code: int = SUCCESS_EXIT_CODE
+        try:
+            set_log_level_from_args(args)
+
+            # If the ssp index file does not exist, create it.
+            ssp_index_path = pathlib.Path(args.ssp_index_path)
+            if not ssp_index_path.exists():
+                # Create a parent directory
+                ssp_index_path.parent.mkdir(parents=True, exist_ok=True)
+
+            ssp_index = SSPIndex(args.ssp_index_path)
+            authored_ssp = AuthoredSSP(args.working_dir, ssp_index)
+
+            comps: List[str] = comma_sep_to_list(args.compdefs)
+            authored_ssp.create_new_default(
+                ssp_name=args.ssp_name,
+                profile_name=args.profile_name,
+                compdefs=comps,
+                markdown_path=args.markdown_path,
+                leveraged_ssp=args.leveraged_ssp,
+                yaml_header=args.yaml_header_path,
+            )
+
+            # The starting point for SSPs is the markdown, so assemble into JSON.
+            model_filter: ModelFilter = ModelFilter([], [args.ssp_name])
+            assemble_task = AssembleTask(
+                authored_object=authored_ssp,
+                markdown_dir=args.markdown_path,
+                version=args.version,
+                model_filter=model_filter,
+            )
+            pre_tasks: List[TaskBase] = [assemble_task]
+
+            super().run_base(args, pre_tasks)
+        except Exception as e:
+            exit_code = handle_exception(e)
+
+        sys.exit(exit_code)
+
+
+def main() -> None:
+    """Run the CLI."""
+    parser = argparse.ArgumentParser(
+        description="Create new system security plan for editing."
+    )
+    create_ssp = CreateSSPEntrypoint(parser=parser)
+
+    args = parser.parse_args()
```

Comment:
not sure if it matters much but maybe setting args should happen before CreateSSPEntrypoint eg. this way simple parser.args errors halt execution before any other 'machinery' is initiated.

Reply:
Thanks for this feedback. The reason `CreateSSPEntryPoint` is created before the parsing the arguments is because the CLI [specific arguments](https://github.com/RedHatProductSecurity/trestle-bot/blob/6b6869f08e7adb259b00f97f3f104af554e6e92a/trestlebot/entrypoints/create_ssp.py#L39) and [shared arguments](https://github.com/RedHatProductSecurity/trestle-bot/blob/34c94341ae57e80beb85f512f2d00c2f7e9c0792/trestlebot/entrypoints/entrypoint_base.py#L36) are adding to the parser during object initialization. We talked about `click` previously which might make this more straightforward, but sticking with `argparse` for now since it is used to also interact with `compliance-trestle`. Still you bring up a great point about how this could be more explicit/expected. With this additional context, do you have any thoughts?

Reply:
no additional thoughts